src/main/kotlin/org/jitsi/jibri/sink/impl/FileSink.kt

```diff
@@ -29,23 +32,18 @@ import java.time.format.DateTimeFormatter
  * are able to append a timestamp to the filename, so we can't give the caller full control
  * over the value anyway.  Because of that I just made the value hard-coded.
  */
-class FileSink(recordingsDirectory: Path, callName: String, extension: String = "mp4") : Sink {
+class FileSink(recordingsDirectory: Path, callName: String) : Sink {
     val file: Path
     init {
-        val suffix = "_${LocalDateTime.now().format(TIMESTAMP_FORMATTER)}.$extension"
+        val suffix = "_${LocalDateTime.now().format(TIMESTAMP_FORMATTER)}.$recordingFormat"
         val filename = "${callName.take(MAX_FILENAME_LENGTH - suffix.length)}$suffix"
         file = recordingsDirectory.resolve(filename)
     }
     override val path: String = file.toString()
-    override val format: String = extension
-    override val options: Array<String> = arrayOf(
-        "-profile:v",
-        "main",
-        "-level",
-        "3.1"
-    )
+    override val format: String = recordingFormat
 
     companion object {
+        val recordingFormat: String by config("jibri.ffmpeg.recording-format".from(Config.configSource))
```

Comment:
This is both the argument to '-f' in ffmpeg and also the file extension? Do those always match?

Reply:
No, for example ffmpeg uses 'matroska' for the muxer but the is usually '.mkv'. I figured the extension is not important and I'd rather have the format configurable

Reply:
You could have them be two separate config parameters, with one value defaulting to the other one.